SzenenSteuerung/module.php

```diff
@@ -117,6 +122,40 @@ public function ApplyChanges()
             }
         }
 
+        //Transfer variableIDs to IDs
+        $targets = json_decode($this->ReadPropertyString('Targets'), true);
+        $variableIDs = [];
+        foreach ($targets as $target) {
+            $variableIDs[$target['VariableID']] = $target['ID'];
+        }
+        $scenes = json_decode($this->ReadAttributeString('SceneData'), true);
+        foreach ($scenes as $index => $scene) {
+            foreach ($scene as $id => $value) {
+                if (array_key_exists($id, $variableIDs) && ($variableIDs[$id] != $id)) {
+                    unset($scenes[$index][$id]);
+                    $scenes[$index][$variableIDs[$id]] = $value;
+                }
+            }
+        }
+        $this->WriteAttributeString('SceneData', json_encode($scenes));
+
+        //Remove unused ids form SceneData
+        $ids = [];
+        $targets = json_decode($this->ReadPropertyString('Targets'), true);
```

Comment:
Belege Variablen nicht doppelt! Hier kannst du (zumindest in der aktuellen Struktur, ggfs. musst du in der Überarbeitung mal schauen) einfach die $targets aus Zeile 126 wiederverwenden.